src/BLE_Client.cpp

```diff
@@ -162,116 +167,116 @@ bool SpinBLEClient::connectToServer() {
         }
         return false;
       }
-      SS2K_LOG(BLE_CLIENT_LOG_TAG, "Reconnecting client");
-      BLERemoteService *pRemoteService = pClient->getService(serviceUUID);
+      SS2K_LOG(BLE_CLIENT_LOG_TAG, "Reconnected client");
+    }
+    /** We don't already have a client that knows this device,
+     *  we will check for a client that is disconnected that we can use.
+     */
+    else {
+      pClient = NimBLEDevice::getDisconnectedClient();
+    }
+  }
 
-      if (pRemoteService == nullptr) {
-        SS2K_LOG(BLE_CLIENT_LOG_TAG, "Couldn't find Service");
-        reconnectTries--;
-        return false;
-      }
+  /** No client to reuse? Create a new one. */
+  if (!pClient) {
+    if (NimBLEDevice::getClientListSize() >= NIMBLE_MAX_CONNECTIONS) {
+      Serial.println("Max clients reached - no more connections available");
+      return false;
+    }
 
-      pRemoteCharacteristic = pRemoteService->getCharacteristic(charUUID);
+    pClient = NimBLEDevice::createClient();
 
-      if (pRemoteCharacteristic == nullptr) {
-        SS2K_LOG(BLE_CLIENT_LOG_TAG, "Couldn't find Characteristic");
-        reconnectTries--;
-        return false;
-      }
+    SS2K_LOG(BLE_CLIENT_LOG_TAG, " - Created client");
 
-      if (pRemoteCharacteristic->canNotify()) {
-        SS2K_LOG(BLE_CLIENT_LOG_TAG, "Found %s on reconnect", pRemoteCharacteristic->getUUID().toString().c_str());
-        reconnectTries = MAX_RECONNECT_TRIES;
-        // VV Is this really needed? Shouldn't it just carry over from the previous connection? VV
-        spinBLEClient.myBLEDevices[device_number].set(myDevice, pClient->getConnId(), serviceUUID, charUUID);
-        spinBLEClient.myBLEDevices[device_number].doConnect = false;
-        pRemoteCharacteristic->subscribe(true, onNotify);
-        postConnect(pClient);
-        return true;
-      } else {
-        SS2K_LOG(BLE_CLIENT_LOG_TAG, "Unable to subscribe to notifications");
-        return false;
-      }
-    } else {  // We don't already have a client that knows this device, we will check for a client that is disconnected that we can use.
-      SS2K_LOG(BLE_CLIENT_LOG_TAG, "No Previous client found");
-      // pClient = NimBLEDevice::getDisconnectedClient();
+    pClient->setClientCallbacks(&myClientCallback, false);
+    /** Set initial connection parameters: These settings are 15ms interval, 0 latency, 120ms timout.
+     *  These settings are safe for 3 clients to connect reliably, can go faster if you have less
+     *  connections. Timeout should be a multiple of the interval, minimum is 100ms.
+     *  Min interval: 12 * 1.25ms = 15, Max interval: 12 * 1.25ms = 15, 0 latency, 51 * 10ms = 510ms timeout
+     */
+    pClient->setConnectionParams(12, 12, 0, 51);
```

Comment:
Might want to delete this line, the default settings are safer. Test first of course.

Reply:
> Might want to delete this line, the default settings are safer. Test first of course.

I'll test it for a while, but I love that it's lightning fast to connect/reconnect, so I'm tempted to keep it.

Reply:
As long as you don't get connection issues it should be fine, might want to consider the timeout at least, 510ms is a bit short, maybe set to 100 (1000ms).

Reply:
40 minutes on the test bed with UDP and websocket logging running full blast and multiple simulated disconnects. Everything is perfect so far. 
![image](https://user-images.githubusercontent.com/17362216/152699452-282a3a06-9e2c-4b48-9691-92c9cf6661ec.png)